src/shared/components/Uploader/Uploader.tsx

```diff
@@ -197,14 +201,51 @@ export const Uploader = ({
 
   const placeholderImgId = isPrimaryUiType ? 'img-filled' : 'img-outlined';
   const placeholderImgSize = isPrimaryUiType ? 32 : 24;
-  const hasSizeError = error === UploadFileError.Size;
-  const hasFormatError = error === UploadFileError.Format;
-  const hasDimensionsError = error === UploadFileError.Dimensions;
+  const [hasSizeError, hasFormatError, hasDimensionsError] = useMemo(
```

Comment:
Curious why `useMemo` is needed here

Reply:
Good question. Neither of my `useMemo`s here are particularly warranted. I think I was trying to troubleshoot reactive behaviour during testing and was misguided into thinking this would solve it. I'll get rid of these.

Reply:
I have had many a `setTimeout` for the exact same reason